npm/private/npm_import.bzl

```diff
@@ -689,8 +690,35 @@ _ATTRS = dicts.add(_COMMON_ATTRS, {
     "custom_postinstall": attr.string(),
     "link_workspace": attr.string(),
     "url": attr.string(),
+    "npmrc": attr.label(),
 })
 
+def _get_npm_auth(rctx, download_url):
+    # Read token from npmrc label
+    if rctx.attr.npmrc:
+        npmrc_path = rctx.path(rctx.attr.npmrc)
+        npmrc = rctx.read(npmrc_path).splitlines()
+        for line in npmrc:
```

Comment:
Hmm. One problem with doing this load and parsing logic in `npm_import` and passing the `npmrc` file forward from `npm_translate_lock` is that every `npm_import` will do duplicate work. Probably better to do the `parse_ini` in `npm_translate_lock` and have `npm_import` take an `npm_auth` attribute that takes the dict that this function returns. You could use a string attribute and pass the `json.encode`d representation of the dict to `npm_import` and then just `json.decode` in that repository rule.

Reply:
Awesome, I have followed your approach but I was not able to use the `json.encode`d dict since it was not useful to have it like so in `npm_import` because I would have to parse the token value again if it was an environment variable (duplicate work). So I've managed to refactor this attribute to be a string and just pass the "final" token value to `npm_import`.

Reply:
nice. that's great.